docs/contributor/arch/019-switch-from-gateways-to-a-central-agent.md

```diff
@@ -0,0 +1,107 @@
+
+- Tail-based sampling
+
+  Scenarios like trace sampling require processing all spans of a trace at one instance, requiring a StatefulSet with sticky trace routing. However, that scenario might not be relevant as this isn't usually done on the backend.
+
+### Gateways Drawbacks
+
+- No uniform push endpoint
+
+  By design, there is one gateway for each signal type that the application must communicate with. Consequently, you must always configure a separate OTLP endpoint for each signal type. Using a uniform URL requires an additional component to route the requests.
+
+- Autoscaling is tricky
+
+  The gateway should provide autoscaling capabilities which are tricky to realize. It must be detected very fast if a scale-out is needed and situations where the backend is overloaded must be excluded from the scale-out.
+
+- Every instance caches the whole cluster state
+
+  Every instance of a gateway must be able to enrich the data of every Pod and with that need to cache all Pod metadata.
+
+- Istio mandatory for load balancing
+
+  Pushing OTLP data from an application to the gateway running with replicas requires load balancing. Unfortunately, a Kubernetes service is acting on L4 while GRPC/HTTP2 is on L7, so an additional load balancer is needed. That can be easily solved by using Istio, which becomes a requirement for applications. However, some applications don't use Istio (StatefulSets often don't) but want to provide OTLP data.
+
+- Istio is mandatory for mTLS
+
+  App to Gateway communication is across nodes and with that should use mTLS, so Istio needs to be used which sometimes is not possible. Also, Istio causes additional overhead like noise filtering and collecting observability data about the service mesh should not rely on the service mesh.
+
+- Cumulativetodelta transformation not supported
+
+  This transformation is mandatory for Dynatrace support and requires that data from a Pod gets processed always by the same instance.
+
+- General misconception
+
+  Gateways are usually running outside the cluster as a first entry point. The cluster should get rid of the data as fast as possible as persistence cannot be provided here (things like Kafka would again run outside). Try to compensate for short-lived hiccups. Otherwise, point back the backpressure to the client and do not try to solve the big picture.
+
+## Proposal
+
+Most of the drawbacks can be solved by running the gateway logic node-local only. Every instance always processes only data of the local node. With that a natural scalability is given and can be extended by vertical scaling capabilities. The "old" gateways will be running in agent mode but still called "OTLP Gateway". Additionally, there will be a new uniform service `telemetry-otlp` forwarding to the node-local entity using the [internalTrafficPolicy: local](https://kubernetes.io/docs/reference/networking/virtual-ips/#internal-traffic-policy) setting. Existing services can stay compatible using the same approach.
```

Comment:
'The "old" gateways will be running in agent mode but still called "OTLP Gateway"' 

Isn't this going to be confusing? I trust you have reasons to keep the old name instead of calling them "OTLP Agent", but I'm concerned about the potential for confusion. 
Are we going to document the components as:

> This is the agent, it's called OTLP Gateway.
> These are the gateways, they are called ...node gateway? ...node-local gateway? They're also based on OTLP btw ;)

Reply:
It will be hard to rename it as we use the term in our API. A pipeline has a agent and a gateway running.. Maybe we can come up with a completely different term, let's see. I think we don't have to decide now on the name